src/service/resources-impl.js

```diff
@@ -218,6 +224,16 @@ export class Resources {
     /** @const @private {!Array<function()>} */
     this.passCallbacks_ = [];
 
+    /** @private {?TreeWalker} */
+    this.prerenderWalker_ = null;
+
+    /** @private {number} */
+    this.prerenderElementsCount_ = 0;
```

Comment:
It seems like we already have `buildAttemptsCount_`

Reply:
`buildAttemptsCount_` is the number of elements we tried to build, whereas this ivar is the number of elements we discovered, and will later check their `prerenderAllowed()` configuration to decide whether we want to build them, or try to discover a new element.
Lmk if you think the naming is confusing

Reply:
My concern is that we're checking `buildAttemptsCount_` in [`grantBuildPermission()`](https://github.com/ampproject/amphtml/pull/21555/files#diff-80b9cea934a00a64f15d14cb656dfad4R561). Should this be checking `prerenderElementsCount_` instead? Or should we just return true in `grantBuildPermission` when `useDocumentOrderPrerendering_`?

Reply:
> Or should we just return true in grantBuildPermission when useDocumentOrderPrerendering_?

sgtm, done.
We can remove `buildAttemptsCount_` when cleaning the experiment